### ISSUE TITLE
process: migrate CI from Kokoro to LUCI

### ISSUE THREAD

**Issue by hyangah** [golang/vscode-go#3533]
Move our primary CI to LUCI, the default CI & trybot infra in the Go project.
That will help us align our test workflow with other Go projects, simplify integration with Gerrit & Relui.

 * In addition to the `go` toolchain and `git`, vscode-go project requires extra tools in the builder image.
    - Node.js (npm)
    - xvfb (xvfb-run) for testing in linux
    - access to the network
 
 Other tools can be built with `go install`  and `npm ci`+`npx` on demand.
 
 * We will also need to figure out how to present the vscode-go integration test output.
    
 * Nice to have:
    Windows/Mac support
    Trybot
    
 cc @golang/release @golang/tools-team

**Comment by dmitshur** [comment]
Adding some relevant notes here.

[CIPD](https://chromium.googlesource.com/infra/luci/luci-go/+/refs/heads/main/cipd/README.md) as used by LUCI builders already has an existing package providing Node.js, and it's used for testing the js/wasm port: https://chrome-infra-packages.appspot.com/p/infra/3pp/tools/nodejs.

For something like xvfb, it might work better to add it to the image (e.g., see [here](https://cs.opensource.google/go/x/build/+/master:env/linux-x86-bookworm/Dockerfile;l=36-61;drc=30d17922cc07a64230259027b7fcd31f57cb747b)).

*-longtest builders have internet access by default (note that `go test -short` isn't expected to need internet, so tests that require internet are expected to skip when testing.Short() is true).

**Comment by findleyr** [comment]
@hyangah moved this to the backlog, since this doesn't seem naturally associated with any release. Please remilestone as appropriate.

**Comment by h9jiang** [comment]
In CL 645116, I saw vscode-go test is triggered against go 1.22 1.23 1.24 and gotip. 

The migration is done. I will mark this as closed.

**Comment by findleyr** [comment]
Great!

Just so I understand, there are quite a few tests in the associated build that are skipped. Is that intended?
https://ci.chromium.org/ui/p/golang/builders/try/x_vscode-go-go1.22-linux-amd64_docker/b8724368980837991217/overview

**Comment by findleyr** [comment]
@h9jiang can you please also update `docs/contributing.md` in the vscode-go repo to remove references to Kokoro, and describe LUCI?

**Comment by h9jiang** [comment]
> Is that intended?

Yes. I believe so. Because vscode-go set go 1.23 in go.mod file so test with version that is lower than 1.23 is skipped. (Otherwise, it will use toochain to auto-upgrade which is the same test as 1.23.)

> can you please also update docs/contributing.md in the vscode-go repo to remove references to Kokoro, and describe LUCI?

Thanks for the reminder, let me re-open this issue and draft the CL for it.

**Comment by h9jiang** [comment]
Just realized there are some extra work needed in vscode-go refs/meta/config branch to add LUCI Result +1 as submission requirement. 

I will create another CL for this. Sorry for closing this issue early.

**Comment by h9jiang** [comment]
We have finished the migration from Kokoro to LUCI. Time to mark this as complete!
